spec/controllers/registrations_controller_spec.rb

```diff
@@ -68,4 +65,66 @@
       end
     end
   end
-end
+
+  describe 'ユーザー更新' do
+    let(:user) do
+      create(:user, email: 'test123456789@test.com', password: 'password12345', password_confirmation: 'password12345', &:confirm)
+    end
+
+    before do
+      @request.env['devise.mapping'] = Devise.mappings[:user]
+      sign_in user
+    end
+
+    context '正常系' do
+      it '正しく値が設定された場合、Home 画面が描画されること' do
+        user_params = attributes_for(:user, email: user.email, password:'password',password_confirmation: 'password', current_password: 'password12345' )
+        put :update, params: { user: user_params }
+        expect(response).to redirect_to root_path
+      end
+
+      it '正しく値が設定された場合(パスワード)、flash メッセージが正しく表示されること' do
+        user_params = attributes_for(:user, email: user.email, password:'password',password_confirmation: 'password', current_password: 'password12345' )
+        put :update, params: { user: user_params }
+        expect(flash[:notice]).to eq 'アカウント情報を変更しました。'
+      end
+
+      it '正しく値が設定された場合(パスワード)、ユーザーの情報が更新されていること' do
+        user_params = attributes_for(:user, email: 'test123456789@test.com', password:'password',password_confirmation: 'password', current_password: 'password12345' )
+        expect do
+          put :update, params: { user: user_params }
+          user.reload
+        end.to change{user.valid_password?("password")}
+      end
```

Comment:
user.reload は検証時に入れると良いかと思いました。

```suggestion
        expect do
          put :update, params: { user: user_params }
        end.to change{user.reload.valid_password?("password")}
      end
```